### PR TITLE
feat(cognito): add `tags` property for user pool

### PR DESCRIPTION
**Pull request by kdnakt**
### Reason for this change

The userPoolTags property existed in the L1 construct but was not present in the L2 construct

### Description of changes

Add the tags property for UserPool

### Description of how you validated changes

Added a unit test and an integration test.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-cognito/lib/user-pool.ts [hunk 1]
```diff
@@ -749,6 +749,12 @@ export interface UserPoolProps {
    * @default - no value
    */
   readonly advancedSecurityMode?: AdvancedSecurityMode;
+
+  /**
+   * The tags applied to the user pool.
+   * @default {} - no tags
+   */
```
Comment (badmintoncryer) [inline packages/aws-cdk-lib/aws-cognito/lib/user-pool.ts:756]:
```suggestion
  /**
   * The tags applied to the user pool.
   *
   * @default - no tags
   */
```

File: packages/@aws-cdk-testing/framework-integ/test/aws-cognito/test/integ.user-pool-tags.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+new CfnOutput(stack, 'user-pool-id', {
+  value: userpool.userPoolId,
+});
+
+new IntegTest(app, 'IntegUserPoolTags', {
+  testCases: [stack],
+});
+
+app.synth();
```
Comment (badmintoncryer) [inline packages/@aws-cdk-testing/framework-integ/test/aws-cognito/test/integ.user-pool-tags.ts:26]:
```suggestion
```

**Review by badmintoncryer — CHANGES_REQUESTED** [review]
Thank you for your contribution! I've added 2 minor comments.

**Comment by kdnakt** [comment]
@badmintoncryer Thank you for the suggestion, I've just committed them.

**Review by badmintoncryer — APPROVED** [review]
Thanks!

**Review by GavinZZ — COMMENTED** [review]
In L2 construct, we don't directly provide `tags` property even though it's supported in the relevant L1 construct.

The typical way of adding tag is through `Tags.of(this).add`. Does this method work for you?

**Comment by kdnakt** [comment]
Thank you, the `Tags.of(this).add` solution worked for me.
Would you recommend that I close this PR now?

**Comment by GavinZZ** [comment]
As a side note, if you use `this`, it will default to result in tags being added at stack level this propagating to all resources created in the stack.

If this is not the ideal solution you could also do `Tags.of(resource)`.

Let’s close this PR as I believe this change is not needed anymore.

**Pull request closed without merging**